src/main/java/uk/gov/hmcts/reform/sscs/messaging/HmcHearingsEventTopicListener.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.reform.sscs.messaging;
+
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusErrorContext;
+import com.azure.messaging.servicebus.ServiceBusException;
+import com.azure.messaging.servicebus.ServiceBusFailureReason;
+import com.azure.messaging.servicebus.ServiceBusProcessorClient;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessageContext;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+@Data
+@SuppressWarnings({
+    "PMD.LawOfDemeter",
+    "PMD.CompareObjectsWithEquals",
+    "PMD.CloseResource"
+})
+public final class HmcHearingsEventTopicListener {
+
+    private final HmcHearingsTopicListenerProperties properties;
+
+    public void receiveMessages() {
+        CountDownLatch countdownLatch = new CountDownLatch(1);
+
+        ServiceBusProcessorClient processorClient = new ServiceBusClientBuilder()
+            .connectionString(properties.getConnectionString())
+            .processor()
+            .topicName(properties.getTopicName())
+            .subscriptionName(properties.getSubName())
+            .processMessage(HmcHearingsEventTopicListener::processMessage)
+            .processError(context -> processError(context, countdownLatch))
+            .buildProcessorClient();
+
+        processorClient.start();
+        try {
+            TimeUnit.SECONDS.sleep(5);
+        } catch (InterruptedException e) {
+            log.error("Client processing interrupted: {}%n ", e.getMessage());
+        }
+        processorClient.close();
```

Comment:
Double check if we want the processor client to close or we just want to run it infinitely in the background so it checks all incoming messages over time as if this closes then it won't check for messages until its reopened.

Reply:
I think what we could have is a class which uses this to listen to all messages every minute or something and process it but for now we should close the resource after we have used it.